### PR TITLE
CCM-2770 update docs for free text comms

### PR DESCRIPTION
**Pull request by mark-r-bjss**
## Summary

Update documentation and sandbox server to support free text communications (global routing config) feature.

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Checklist
* [x] Brief description of work completed, and any technical decisions made as part of the PR
* [x] PR link added as a comment to the relevant JIRA ticket
* [x] Branch deployed to a dynamic env - link to deployment pipeline
* [x] PR link shared on Slack and/or Teams
* [x] 2 reviews received
* [ ] Tester approval

### CODE REVIEW

File: sandbox/handlers/utils.js [hunk 1]
```diff
@@ -37,7 +37,24 @@ function sendError(res, code, message) {
...
+    if (!personalisation) {
+      return false;
+    }
+  
+    const personalisationFields = Object.keys(personalisation);
+    if (personalisationFields.length != 1) {
+      return false;
+    }
+  
+    if (personalisationFields[0] !== "body") {
```
Comment (aaron-downing) [inline sandbox/handlers/utils.js:50]: Should we not just be checking if personalisationFields length is greater than 0 and includes `"body"`?

Comment (aaron-downing) [reply sandbox/handlers/utils.js:50]: I'm also on the fence of it strictly checking for "body" when this might change between clients

Comment (aaron-downing) [reply sandbox/handlers/utils.js:50]: Oh, I see this is for the sandbox environment so presumably this is for a dummy request... nevermind

Comment (mark-r-bjss) [reply sandbox/handlers/utils.js:50]: This validation ensures that the personalisation is exactly `{ body: string }`, with `body` being mandatory and no extra redundant fields.  This aims to help sandbox integrators to correctly use the global routing config.

File: sandbox/handlers/config.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ const trigger425SendingGroupId = "d895ade5-0029-4fc3-9fb5-86e1e5370854";
 const sendingGroupIdWithMissingTemplates = "c8857ccf-06ec-483f-9b3a-7fc732d9ad48";
 const sendingGroupIdWithDuplicateTemplates = "a3a4e55d-7a21-45a6-9286-8eb595c872a8";
 const sendingGroupIdWithMissingNHSTemplates = "aeb16ab8-cb9c-4d23-92e9-87c78119175c";
+const globalNhsAppSendingGroupId = "00000000-0000-0000-0000-000000000001";
```
Comment (harrim91) [inline sandbox/handlers/config.js:7]: I don't know if we would have any other global nhsapp templates, but may be good to add free-text into this variable name

File: sandbox/handlers/batch_send.js [hunk 3]
```diff
@@ -1,150 +1,192 @@
...
+    return;
+  }
+
+  if (!req.body) {
+    sendError(res, 400, "Missing request body");
+    next();
+    return;
+  }
+
+  if (!req.body.data) {
```
[the remark this answers is not in the document]
Comment (mark-r-bjss) [reply sandbox/handlers/batch_send.js]: Done.

File: sandbox/handlers/config.js (hunk 2 above)
Comment (mark-r-bjss) [reply sandbox/handlers/config.js]: Done.

**Review by aaron-downing — APPROVED** [review]

**Review by harrim91 — APPROVED** [review]

**Review by TheWorldWideWebb — APPROVED** [review]

**Comment by mark-r-bjss** [comment]
> 1. AssertionError  Status code is 201
>    expected response to have status code 201 but got 404
>    at assertion:1 in test-script
>    inside "Send Message Batches / 201 - Message Batch Created - Free-text NHS App"
> 2. AssertionError  Status code is 201
>    expected response to have status code 201 but got 404
>    at assertion:1 in test-script
>    inside "Send Message / 201 - Message Created - Free-text NHS App"
>    ```
>                Think theres some test issues for internal sandbox ?
>    ```

I'm planning to split the postman collection change into a separate ticket because it needs sandpit to support the new sendingGroupId in order for the postman tests to pass.

**Review by TheWorldWideWebb — APPROVED** [review]

**Pull request merged**